### PR TITLE
Adds kubefed dns-provider validation

### LINKED ISSUE #48732: kubefed should emit error when passing invalid dns-provider
/kind bug

**What happened**:
I've misspelled the dns-provider (put googlecloud-dns instead of google-clouddns) when running `kubefed init`
It successfully ran and said everything was ok. However, federation-control-plane pod entered in CrashLoopBackOff and I could only see the error in the pod's logs:
```F0710 20:15:37.946633       1 controllermanager.go:174] Cloud provider could not be initialized: unknown DNS provider "googlecloud-dns"```

**What you expected to happen**:
If any string different than 'google-clouddns', 'aws-route53' or 'coredns' is provided in `dns-provider`, kubefed should return an error to the client.

**Environment**:
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-06-29T23:15:59Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

### PR DESCRIPTION
**Pull request by henriquetruta**
Adds a validation for kubefed only accept one valid
dns-provider (google-clouddns, aws-route53, or coredns)

fixes #48732

```release-note
dns-provider name validation
```

### CODE REVIEW

File: federation/pkg/kubefed/init/init.go [hunk 1]
```diff
@@ -125,6 +125,10 @@ var (
 		"app":    "federated-cluster",
 		"module": "federation-controller-manager",
 	}
+
+	validDNSProviders = []string{
```
Comment (nikhiljindal) [inline federation/pkg/kubefed/init/init.go]: Have you tried using RegisteredDnsProviders?
Example usage: https://github.com/kubernetes/kubernetes/blob/5af81b0955d907256a222e5cbcccfcba0db88b53/federation/cmd/federation-controller-manager/app/options/options.go#L126

**Comment by nikhiljindal** [comment]
/ok-to-test.

@henriquetruta Please add a test as well (which verifies that the expected error is returned with an invalid dns provider)

**Comment by nikhiljindal** [comment]
Also add a release note since this is a user visible change

File: federation/pkg/kubefed/init/init.go (hunk 1 above)
Comment (henriquetruta) [inline federation/pkg/kubefed/init/init.go]: Just looked at it. But seems like it's broken on the tests. It only returns [coredns]

**Pull request closed without merging**